substrate/frame/support/src/traits/tokens/misc.rs

```diff
@@ -299,6 +300,33 @@ where
 	fn ensure_successful(_: AssetId) {}
 }
 
+/// Implements [`ConversionFromAssetBalance`], allowing for a 1:1 balance conversion of the asset
+/// when it meets the conditions specified by `C`. If the conditions are not met, the conversion is
+/// delegated to `O`.
+pub struct UnityOrOuterConversion<C, O>(core::marker::PhantomData<(C, O)>);
```

Comment:
```suggestion
pub struct OneToOneIfElse<C, O>(core::marker::PhantomData<(C, O)>);
```

or

```suggestion
pub struct OneToOneOrOuterConversion<C, O>(core::marker::PhantomData<(C, O)>);
```

I think it expresses what you want better

Reply:
we have type above with a name - `UnityAssetBalanceConversion`, which also means 1:1. its not wrong and I would keep it consistent.